preprocessing/structs.ts

```diff
@@ -96,121 +161,159 @@ const id_field = (name: string) => {
+      { name: "maxGasreqForFailingOffers", bits: 32, type: "uint" },      
+    ],
+  },
+
+  /* ## Local configuration */
+  local: {
+    fields: [
+      /* * An offer list is not `active` by default, but may be activated/deactivated by governance. */
+      { name: "active", bits: 1, type: "bool" },
+      /* * `fee`, in basis points, of `outbound_tkn` given to the taker. This fee is sent to Mangrove. Fee is capped to ~2.5%. */
+      { name: "fee", bits: 8, type: "uint" },
+      /* * `density` is similar to a 'dust' parameter. We prevent spamming of low-volume offers by asking for a minimum 'density' in `outbound_tkn` per gas requested. For instance, if `density` is worth 10, `offer_gasbase == 5000`, an offer with `gasreq == 30000` must promise at least _10 × (30000 + 5000) = 350000_ `outbound_tkn`. _9 bits wide_.
+
+      We store the density as a float with 2 bits for the mantissa, 7 for the exponent, and an exponent bias of 32, so that density ranges from $2^{-32}$ to $1.75 \times 2^{95}$. For more information, see `DensityLib`.
+      
+      */
+      { name: "density", bits: 9, type: "Density", underlyingType: "uint"},
+      /* To save on gas, Mangrove cache the entire tick tree branch of the bin that contains the best offer in each offer list's `local` parameter. Taken together, `binPosInLeaf`, `level3`, `level2`, `level1` and `root` provide the following info:
+      - What the current bin is (see `BinLib.bestBinFromLocal`)
+      - When a leaf is emptied and the next offer must be fetched, the information in the fields `level3`, `level2`, `level1` and `root` avoid multiple storage reads
+      */
+      { name: "binPosInLeaf", bits: 2, type: "uint" },
+      { name: "level3", bits: 64, type: "Field", underlyingType: "uint" },
+      { name: "level2", bits: 64, type: "Field", underlyingType: "uint" },
+      { name: "level1", bits: 64, type: "Field", underlyingType: "uint" },
+      { name: "root", bits: 2, type: "Field", underlyingType: "uint" },
+      /* * `offer_gasbase` represents the gas overhead used by processing the offer inside Mangrove + the overhead of initiating an entire order. Mangrove considers that a failed offer has used at least `offer_gasbase` gas. The actual field name is `kilo_offer_gasbase` and the accessor `offer_gasbase` returns `kilo_offer_gasbase*1e3`. Local to an offer list, because the costs of calling `outbound_tkn` and `inbound_tkn`'s `transferFrom` are part of `offer_gasbase`. Should only be updated when ERC20 contracts change or when opcode prices change. */
+      fields.kilo_offer_gasbase,
+      /* * If `lock` is true, orders may not be added nor executed, nor the offer list read by external contracts.
+
+        Reentrancy during offer execution is not considered safe:
```

Comment:
Maybe we should also mention, that it may be possible to detect cleaning.

Reply:
We should explain that if we didn't have the read-lock it might be possible for malicious makers to detect cleaning and thus avoid it.